sqlitecluster/SQLite.cpp

```diff
@@ -208,18 +212,82 @@ void SQLite::_sqliteLogCallback(void* pArg, int iErrCode, const char* zMsg) {
 
 int SQLite::_sqliteWALCallback(void* data, sqlite3* db, const char* dbName, int pageCount) {
     SQLite* object = static_cast<SQLite*>(data);
-    int autocheckpoint = object->_checkpointInterval;
-    if (autocheckpoint) {
-        if (pageCount < autocheckpoint) {
-            SINFO("WAL callback skipping checkpoint with " << pageCount << " pages in WAL file.");
+    // Try a passive checkpoint if full checkpoints aren't enabled, *or* if the page count is less than the required
+    // size for a full checkpoint.
+    if (!object->_enableFullCheckpoints || pageCount < fullCheckpointPageMin.load()) {
+        int passive = passiveCheckpointPageMin.load();
+        if (!passive || pageCount < passive) {
+            SINFO("[checkpoint] skipping checkpoint with " << pageCount
+                  << " pages in WAL file (checkpoint every " << passive << " pages).");
         } else {
             int walSizeFrames = 0;
             int framesCheckpointed = 0;
             uint64_t start = STimeNow();
             sqlite3_wal_checkpoint_v2(db, dbName, SQLITE_CHECKPOINT_PASSIVE, &walSizeFrames, &framesCheckpointed);
-            SINFO("WAL callback passive checkpoint with " << pageCount << " pages in WAL file. Frames checkpointed: "
-            << framesCheckpointed << " of " << walSizeFrames << " in " << ((STimeNow() - start) / 1000) << "ms.");
+            SINFO("[checkpoint] passive checkpoint complete with " << pageCount
+                  << " pages in WAL file. Frames checkpointed: " << framesCheckpointed << " of " << walSizeFrames
```

Comment:
do you want to fix framesCheckpointed to be the actual value of the checkPointed frames? i.e. exclude the previous ones.

Reply:
No, because I don't want to keep track of what it was previously. Is there a clearer wording we can use for the log line with the existing numbers?